src/plugins/discover/public/application/helpers/get_context_url.tsx

```diff
@@ -19,26 +19,33 @@
 import { stringify } from 'query-string';
 import rison from 'rison-node';
 import { url } from '../../../../kibana_utils/common';
-import { esFilters, FilterManager } from '../../../../data/public';
+import { esFilters, FilterManager, TimeRange } from '../../../../data/public';
 
 /**
  * Helper function to generate an URL to a document in Discover's context view
  */
-export function getContextUrl(
-  documentId: string,
-  indexPatternId: string,
-  columns: string[],
-  filterManager: FilterManager
-) {
+export function getContextUrl(props: {
+  documentId: string;
+  indexPatternId: string;
+  columns: string[];
+  filterManager: FilterManager;
+  timeRange: TimeRange;
+  /* Routing is optional, but allows the context query to be more optimized
+   */
+  routing: string | undefined;
```

Comment:
The most important question to answer here is: are these the right 2 parameters, or should we be querying the concrete index instead?

Reply:
if we would query the the concrete index this could certainly speed up search in certain cases, however, we would need to make the `index` optional, to stay compatible, since currently we're using index pattern. I think `_routing` could certainly help to find the anchor document in a faster way when given. I'm not sure adding the time range helps in terms of speeding up the `ids` query, which I expected to be already fast? Would that bring any benefit, dear @nik9000?

Reply:
If you are targeting a single index the time range won't do anything. It'll make it ever so slightly slower, I think. But if you are targeting many indices it'll let us pick only shards that can have the document.

Reply:
thx @nik9000

Reply:
We are using the index pattern, not the concrete index. I've seen a few examples where this matters, for example if you use index aliases but don't have permission for the concrete indices. So I think this validates the approach I'm doing here of using the index pattern + time filter.

Reply:
for sure @wylieconlon , it's fine to move forward in this direction 👍